promises/mod.ts

```diff
@@ -32,11 +32,26 @@ export function defer<T>(): Deferred<T> {
   };
 }
 
-export function isDeferred(x): x is Deferred {
-  return (
-    typeof x === "object" &&
-    x.promise instanceof Promise &&
-    typeof x["resolve"] === "function" &&
-    typeof x["reject"] === "function"
-  );
+export function delay<T>(ms: number, value?: T): Promise<T> {
+  return new Promise<T>((resolve, reject) => {
+    setTimeout(() => {
+      resolve(value);
+    }, ms);
+  });
+}
+
+export class TimeoutError extends Error {
+  constructor(ms: number) {
+    const message = `Timeout after ${ms}ms`;
+    super(message);
```

Comment:
Could these two lines be combined?

```suggestion
    super(`Timeout after ${ms}ms`);
```

Also, should this use something like Node’s [`ms`](https://npm.im/ms) utility to display the elapsed time in a more friendly manner?